### DISCUSSION TITLE
just a quick question/idea

### DISCUSSION THREAD

**Discussion by otalado** [dail8859/NotepadNext#255] [Ideas]
From time to time it'll be very handy to be able to view/open hidden folders/files

**Comment by dail8859** [comment]
As far as I know Qt uses the OS's native file dialog. On Windows I did a quick test and it follows whatever setting you have in Explorer whether to show hidden files or not.

**Reply by otalado** [reply]
I tested that on Raspberry Pi OS (Bullseye) and on Debian Bullseye (x64). I have changed the default value with dconf-editor to show hidden files/folders in open file dialogs. It turns out that I now see the hidden files/folders in MousePad (as expected) but, unfortunately not in NotepadNext.

**Reply by dail8859** [reply]
Thanks for the info. Looks like [`QDir::Hidden`](https://doc.qt.io/qt-5/qdir.html#Filter-enum) might be what needs added to the Open File dialog to properly show them on Linux.

Are you using the AppImage for flatpak?

**Reply by dail8859** [reply]
I've made a change I think will fix this.

If you are using the AppImage (I'm not sure if that even works on Raspberry Pi?) you can download it here: https://github.com/dail8859/NotepadNext/suites/9121696284/artifacts/423461266

Just unzip it then `chmod +x` to run it.

If you are using the flatpak install I can push out a beta release to make sure it fixes it for you.

**Reply by otalado** [reply]
Could not test on Raspberry Pi as the zip contains x86_64 AppImage. But I tested it on an intel pc with debian 11. Sadly it does not show hidden files/folders.

**Reply by otalado** [reply]
Oh, sorry, I forgot to reply: on RPI I used the flatpack to install NotepadNext

**Reply by dail8859** [reply]
> But I tested it on an intel pc with debian 11. Sadly it does not show hidden files/folders.

I tried the AppImage on Ubuntu and it does show the hidden files. Not sure why it isn't working for you but I'll go ahead and merge in the pull request since this functionality is desirable in the cases it does work.

**Reply by NichtsHsu** [reply]
Although NotepadNext didn't show the hidden files on my Ubuntu 22.04 by default, I can switch it simply by pressing `Ctrl+H`. So I believe this should not be a serious issue for Ubuntu users, and users of other Linux systems can also try to see if it is available.

Defaultly:

![image](https://user-images.githubusercontent.com/35688135/200723459-2a2a8f30-5c85-4667-8705-4215dc13b7f8.png)

After `Ctrl+H`:

![image](https://user-images.githubusercontent.com/35688135/200723482-fe95160a-7da8-4c4d-870c-80d7353546fa.png)

**Reply by dail8859** [reply]
@NichtsHsu Good point. I just tested this on Ubuntu 21.10 and I'm able to Right-Click and show the hidden files. I did notice that it needed set every time the dialog opens...which is definitely not ideal.

From a code standpoint I'd love to remove the code I added for enabling hidden files by default (had to copy code from the Qt libraries), but from a user perspective controlling this from the application side provides better consistency.
